python/bot/operator_bot.py

```diff
@@ -47,6 +47,9 @@ def create_operator(event):  # pylint: disable=unused-variable
             return client.submit_create(CHAT.Operator, { 'operator': client.party })
         else:
             logging.info(f'Operator {party} is ready')
```

Comment:
what is the net effect of ACKing the existing sessions? do sessions ever get invalidated? what is the (presumably positive) side effect of the ack that motivates this change more broadly?

Reply:
acking kicks off some business logic in daml that decides whether to invite the user to the app (new user) or return their user contract (if they are a returning user). The session contract only exists to trigger that logic (which is a workflow initiated from the client), then it gets archived. In the future we may change this (there is no way right now for knowing when the user logged off). An integration that works with time (like setting up a timer) would be a good first step to use in order to keep the contract around and then invalidate it after x amount of time

Reply:
do sessions ever get invalidated? wouldn't this result in a linearly increasing number of sessions over time? why is it necessary to find all of the active ones in order to ack, instead of just the new one?

Reply:
'sessions' in this case are short lived. they get acked  (and archived) as soon as they get created by the client's browser. The sessions increase linearly by the number of users. Finding the active ones means that the client's browser has created one and the bot was not listening then. so upon startup it just gathers all past usersessions and acks them